### PR TITLE
Added Roles and Security

### PR DESCRIPTION
**Pull request by mhorianin**

### CODE REVIEW

File: src/main/java/com/example/bookstoreweb/model/User.java [hunk 1]
```diff
@@ -2,16 +2,25 @@
...
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
 import lombok.Data;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 @Data
 @Entity
 @Table(name = "users")
```
Comment (okuzan) [inline src/main/java/com/example/bookstoreweb/model/User.java:26]: user should implement soft delete

File: src/main/java/com/example/bookstoreweb/model/User.java [hunk 2]
```diff
@@ -30,4 +39,44 @@ public class User {
...
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return true;
```
Comment (okuzan) [inline src/main/java/com/example/bookstoreweb/model/User.java]: should depend on if it's deleted

File: src/main/resources/db/changelog/changes/03-create-roles-table.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: role_name
+                  type: ENUM('USER', 'ADMIN')
+                  constraints:
+                    unique: true
+                    nullable: false
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/03-create-roles-table.yaml]: add empty line

File: src/main/resources/db/changelog/changes/09-set-bcrypt-password.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+databaseChangeLog:
+  - changeSet:
+      id: set-bcrypt-password
+      author: mhorianin
+      changes:
+        - update:
+            tableName: users
+            columns:
+              - column: {name: password, value: '$2a$10$zarKKAjobZ6jX51HiSBl7eNFX4BeZ9Ch7EqulRX9TK/4Soxz5NntW'}
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/09-set-bcrypt-password.yaml:9]: add comment with password in plaintext

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookstoreweb/controller/BookController.java [hunk 5]
```diff
@@ -27,31 +28,36 @@
...
     private final BookService bookService;
 
+    @PreAuthorize("hasAuthority('USER')")
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of all books")
     public List<BookDto> getAll(Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
+    @PreAuthorize("hasAuthority('USER')")
```
Comment (olekskov) [inline src/main/java/com/example/bookstoreweb/controller/BookController.java]: Are you sure admin shouldn't have ability to get books?

**Review by olekskov — APPROVED** [review]

**Pull request merged**